bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/type/HomematicTypeGeneratorImpl.java

```diff
@@ -301,11 +303,10 @@ public EventOption createOption(String value, String description) {
                                 return new EventOption(value, description);
                             }
                         }));
-
             }
             channelType = new ChannelType(channelTypeUID, !MetadataUtils.isStandard(dp), itemType, channelKind, label,
```

Comment:
While you are already touching the code, may I ask you to replace this constructor by [ChannelTypeBuilder](https://github.com/openhab/openhab-core/blob/3811cb95604f69f6a3d77da2caf391d81b05458e/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/type/ChannelTypeBuilder.java) too?

Reply:
I have expected this comment :-). Unfortunately I could not simply replace it by the ChannelTypeBuilder because the channel type `channelKind` is defined as enum here and the constructor call is used to create a *State* channels as well as *Trigger* channels. This means some additional changed and I preferred to do this in a separate step to get rid of these deprecated constructors as soon as possible.

Reply:
I get your point and of course it is an improvement but the thing is that exactly this ctor will be removed in https://github.com/openhab/openhab-core/pull/1414 and thus this PR will not fix the build of the OH3 add-on port.

Isn't this code snippet a simple solution:

```java
                        }));
                channelType = ChannelTypeBuilder.trigger(channelTypeUID, label).with....build();
            }
            channelType = ChannelTypeBuilder.state(channelTypeUID, label, itemType).with....build();
        }
```

Reply:
OK, this constructor is currently not marked as deprecated. Therefore I thought I could use it. 
I have changed it.